File: Swift-C++-IR-Comparison/test/structsAndClasses/Swift/StructChangeSelf.swift

import Foundation
@main
struct MyMain {

    static func main() {
        var somePoint = Point(x: 1.0, y: 1.0)
        somePoint.moveBy(x: 2.0, y: 3.0)
        let newPosition = (somePoint.x, somePoint.y)
    }
}
struct Point {
    var x = 0.0, y = 0.0
    mutating func moveBy(x deltaX: Double, y deltaY: Double) {
        self = Point(x: x + deltaX, y: y + deltaY)
    }
}

File: Swift-C++-Testsuite/test/structsAndClasses/C++/Structs/StructChangeSelf.cpp

 #include <iostream>

struct Point {
    double x;
    double y;
    void moveBy(const double deltaX, const double deltaY) {
    x += deltaX;
    y += deltaY;
}
};



int main() 
{
    Point somePoint;
    somePoint.x = 1.0;
    somePoint.y = 1.0;

    somePoint.moveBy(2.0, 3.0);
    std::cout << somePoint.x << ", " << somePoint.y;
    return 0;
}